telosnetwork/telos-wallet
# #603 | Allowances table

# Fixes #603 

## Description
This PR adds the UI (with mock data) for the allowances table, as well as:
- refactors the NFT store slightly so you can now directly fetch all NFTs for a collection, rather than having to get NFTs for a user in order to get them for a collection
- prevents the NFT class from getting the first frame for video NFTs - this was already not working, but it was causing huge numbers of videos to be loaded over the network to attempt to do so. This is because for every video NFT (which there may be hundreds of in a given collection), an offscreen `<video>` element was being rendered so the first frame could be extracted and used as a thumbnail. this functionality wasn't actually working so removing that step is only saving bandwidth/time
- moves some date utils to the `date-utils` file from the generic utils file
- updates a11y for table controls

For this PR, it's important to understand the different types of data we need to show. In a nutshell, an allowance row can represent:
- an ERC20 allowance
- an allowance for a single ERC721 NFT
- an allowance for all ERC721 NFTs in a given collection
- an allowance for all ERC1155 NFTs in a given collection

Each of these types of rows has slightly different data. I did my best to represent different permutations of info with the mock data. Let me know if you think i missed a scenario, or if anything doesn't make sense 

## Test scenarios
- run `git fetch --all && git checkout 603-allowances-table && yarn dev`
- go to http://localhost:8081
- sign in
- go to allowances page
    - you should see the allowances table
- hover over table cells in the Asset, Value, and Allowance columns
    - appropriate tooltips should show on hover 
- mess with the pagination
    - the table should behave as expected

## Checklist:
